### PR TITLE
Add support for missing geospatial datatypes to SQLAlchemy dialect

### PR DESCRIPTION
**Pull request by autophagy**
This should address #273

### CODE REVIEW

File: docs/sqlalchemy.rst [hunk 1]
```diff
@@ -284,6 +284,58 @@ The resulting object will look like this::
...
     ``ObjectArray`` data.
 
+.. _geopoint:
+.. _geoshape:
+
+``Geopoint`` and ``Geoshape``
+.............................
+
+The CrateDB SQLAlchemy dialect provides two geospatial types:
+- ``Geopoint``, which represents a longitude and latitude coordinate
```
Comment (mechanomi) [inline docs/sqlalchemy.rst:294]:
```suggestion

- ``Geopoint``, which represents a longitude and latitude coordinate
```

File: docs/sqlalchemy.rst [hunk 2]
```diff
@@ -284,6 +284,58 @@ The resulting object will look like this::
...
+- ``Geoshape``, which is used to store geometric `GeoJSON geometry objects`_.
```
Comment (mechanomi) [inline docs/sqlalchemy.rst:295]:
```suggestion
- ``Geoshape``, which is used to store geometric `GeoJSON geometry objects`_
```

File: docs/sqlalchemy.rst [hunk 3]
```diff
@@ -284,6 +284,58 @@ The resulting object will look like this::
...
+
+To use these types, you can create columns using them::
```
Comment (mechanomi) [inline docs/sqlalchemy.rst:297]:
```suggestion
To use these types, you can create columns, like so::
```

File: docs/sqlalchemy.rst [hunk 4]
```diff
@@ -284,6 +284,58 @@ The resulting object will look like this::
...
+    ...    name = sa.Column(sa.String, primary_key=True)
+    ...    coordinate = sa.Column(types.Geopoint)
+    ...    area = sa.Column(types.Geoshape)
+
+To create a geopoint, you can either define it as a tuple of
+``(longitude, latitude)``::
+
+    >>> point = (139.76 35.68)
+
+or as a geojson ``Point`` object::
```
Comment (mechanomi) [inline docs/sqlalchemy.rst:311]: it's generally not considered good style to split a sentence up with an example. can you reword so each leading sentence is complete?

File: src/crate/client/doctests/sqlalchemy.txt [hunk 5]
```diff
@@ -182,6 +182,53 @@ all values of that field of all objects in that object array::
...
+    ...             [139.806, 35.515],
+    ...             [139.919, 35.703],
+    ...             [139.768, 35.817],
+    ...             [139.575, 35.760],
+    ...             [139.584, 35.619],
+    ...             [139.806, 35.515],
+    ...         ]
+    ...     ]
+    ... )
+    >>> point = Point((139.76, 35.68))
```
Comment (MarkusH) [inline src/crate/client/doctests/sqlalchemy.txt:216]: Above we do this. Should we keep it consistent?
```suggestion
    >>> point = Point(coordinates=(139.76, 35.68))
```

File: src/crate/client/sqlalchemy/types.py [hunk 6]
```diff
@@ -217,3 +220,50 @@ def get_col_spec(self, **kws):
...
+        def process(value):
+            if isinstance(value, geojson.Point):
+                value = value.coordinates
+            return value
+        return process
+
+    def result_processor(self, dialect, coltype):
+        def process(v):
+            return tuple(v)
+        return process
```
[the remark this answers is not in the document]
Comment (autophagy) [reply src/crate/client/sqlalchemy/types.py:247]: Ah! Yeah, you're completely right. Good spot!

**Comment by autophagy** [comment]
@MarkusH I've pushed up a commit that should hopefully address your comments!

**Pull request merged**